internal/grpc/commander/connection.go

```diff
@@ -93,19 +249,123 @@ func (c *connection) receive(ctx context.Context) error {
 	}
 }
 
+func (c *connection) updateConfigLoop(ctx context.Context) error {
+	defer func() {
+		c.logger.Info("Stopping update config loop")
+	}()
+	c.logger.Info("Starting update config loop")
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-c.configUpdatedCh:
+			c.waitForConfigApply(ctx)
+		}
+	}
+}
+
+func (c *connection) waitForConfigApply(ctx context.Context) {
+	config := c.observedConfig.GetLatestConfig()
+	if config == nil {
```

Comment:
It depends on how you view the fact that there's no config stored. IMO, that's not necessarily an error, it could mean that the controller hasn't processed any gateway resources yet, so there's no config to generate.